build_tools/python/e2e_test_framework/models/matmul.py

```diff
@@ -0,0 +1,69 @@
+## Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Defines Matmul microbenchmarks."""
+
+from e2e_test_framework import unique_ids
+from e2e_test_framework.definitions import common_definitions
+
+MATMUL_3456X1024X2048_FP16_MLIR = common_definitions.Model(
+    id=unique_ids.UB_MATMUL_3456X1024X2048_FP16_MLIR,
+    name="matmul_3456x1024x2048_f16t_f16t_f16t_tile_config_default",
+    tags=["fp16", "ubench", "matmul"],
+    source_type=common_definitions.ModelSourceType.EXPORTED_LINALG_MLIR,
+    source_url=
+    "https://storage.googleapis.com/iree-model-artifacts/microbenchmarks/matmul/matmul_3456x1024x2048_f16t_f16t_f16t_tile_config_default.mlir",
```

Comment:
This needs a unique ID (e.g. the hash) since it's using an unstable source format. Can we also use mlir bytecode here instead of text? That would make the extension .mlirbc I think

Reply:
The source URL needs a hash? This might need larger changes to the framework. I think it's useful to have the microbenchmark mlir readable straight away. They're very small files anyway so I don't see the need to convert it to binary.

Reply:
Yeah the source URL should have a hash because in order to update it atomically with source changes, someone needs to be able to change the name. There's nothing really special about hashes: it just gives us a unique name.

The reason for using the bytecode format is primarily stability, not size. It should be very easy to convert to/from bytecode (if it's not, we have a problem).

Reply:
Instead of a hash, can we use a date timestamp? This would be more readable and easier to reason about in terms of directory structures.

Reply:
Sure. It means someone can't have two attempts on the same day though :-) We use a digest (note I mean appended to the name, not instead of the name) in a couple of these already, so I'd prefer to keep them consistent, but if you feel strongly about dates instead then I'm OK with that (though maybe you could update the other IR-based benchmark to also use a date?). One other thing is that GCS should show you the file modified time, so that metadata should be visible somewhere still.

Reply:
I don't have a strong opinion on the format. Can you point me to an example of the digest and how to generate it?

Reply:
Well I went looking for the previous example of this and remembered that it was actually for an e2e test and that apparently the PR was never merged :grimacing:. Here's the README for that (which should be merged shortly, pending presubmits): https://github.com/sogartar/SHARK-Runtime/blob/794df86f6a440dcca35563964744e456675a76d3/tests/e2e/models/mnist_train_test/README.md

I see that actually the best analogue for what you're doing is for the torch models (which I was mixing up with the above) where you've used dates: https://github.com/openxla/iree/blob/df7522b098f9be1252a114d76a80c5fba9b19d42/build_tools/python/e2e_test_framework/models/torch_models.py#L29

So my argument about consistency is shot :stuck_out_tongue: Do whatever you think best. The important thing to me is just that someone can make atomic source code changes and that they don't have to do out-of-band things too often.

Reply:
I created a directory with a datetime and converted the mlir files to mlirbc